code/modules/mob/living/living_say.dm

```diff
@@ -231,7 +231,14 @@ GLOBAL_LIST_INIT(message_modes_stat_limits, list(
 		message = "[randomnote] [message] [randomnote]"
 		spans |= SPAN_SINGING
 
+	#ifdef UNIT_TESTS
+	// Saves a ref() to our arglist specifically.
+	// We do this because we need to check that COMSIG_MOB_SAY is getting EXACTLY this list.
+	last_say_args_ref = REF(args)
+	#endif
+
 	// Leaving this here so that anything that handles speech this way will be able to have spans affecting it and all that.
+	// Make sure the arglist is passed exactly - don't pass a copy of it. Say signal handlers will modify some of the parameters.
 	var/sigreturn = SEND_SIGNAL(src, COMSIG_MOB_SAY, args, message_range)
```

Comment:
Appending args by adding them to the end of the parameters does not work.  I tested this on tg master (not my hivemind branch)

![Code_yXIlmidcjY](https://user-images.githubusercontent.com/5195984/189693736-d3988125-22d2-4b1a-b74d-0cbf55defdea.png)

The `message_range` arg does not get added to args.  It gets thrown away.